mirrord/layer/src/socket/ops.rs

```diff
@@ -108,6 +108,30 @@ pub(super) fn socket(domain: c_int, type_: c_int, protocol: c_int) -> Detour<Raw
     Detour::Success(socket_fd)
 }
 
+#[tracing::instrument(level = "warn", ret)]
+fn bind_port(sockfd: c_int, domain: c_int, port: u16) -> Detour<()> {
+    let address = match domain {
+        libc::AF_INET => Ok(SockAddr::from(SocketAddr::new(
+            IpAddr::V4(Ipv4Addr::LOCALHOST),
+            port,
+        ))),
+        libc::AF_INET6 => Ok(SockAddr::from(SocketAddr::new(
+            IpAddr::V6(Ipv6Addr::UNSPECIFIED),
+            port,
+        ))),
+        invalid => Err(Bypass::Domain(invalid)),
+    }?;
+
+    trace!("bind -> {address:#?}");
+
+    let bind_result = unsafe { FN_BIND(sockfd, address.as_ptr(), address.len()) };
+    if bind_result != 0 {
+        return Detour::Error(io::Error::last_os_error().into());
+    }
+
+    Detour::Success(())
```

Comment:
```suggestion
    if bind_result != 0 {
        Detour::Error(io::Error::last_os_error().into())
    } else {
        Detour::Success(())
    }
```

Considering that this is the last bit of the function, wdyt?